copy/single.go

```diff
@@ -398,6 +400,29 @@ func (ic *imageCopier) compareImageDestinationManifestEqual(ctx context.Context,
 	}, nil
 }
 
+// canSkipValidation returns true if the source supports storing partial images and the image is.
+func canSkipValidation(ctx context.Context, annotations map[string]string, src private.ImageSource) (bool, error) {
+	toc, err := toc.GetTOCDigest(annotations)
+	if err != nil {
+		return false, err
+	}
+	if toc == nil {
+		return false, nil
+	}
```

Comment:
How does the presence of the TOC really factor into the decision here?

I appreciate that the actual decision mechanism below hasn’t been designed yet… my first guess “the source transport claims that the blob is valid and doesn’t need verifying” would just as well apply to TOC-less layers. Or maybe the source transport should be returning a per-blob value instead of a global constant? (But how would c/storage make that decision?)

Reply:
the presence of the TOC is to somehow limit the `skipValidation` only to images that could possibly be stored using a partial pull.  Of course, this is not a security measure because a malicious user could just use the annotation and inhibit the check we have in place.

That is why I am using the check `canSkipValidation`, to skip the validation only when we are copying from transports we can trust.  We can make it explicit in the transport itself and declare for each transport if it is safe to trust its content.  In practice, we would do it only for the `containers-storage` transport.

Is it a safe assumption?  Are there cases where we cannot trust what we have in the storage?

Reply:
This was removed, now the sources provide `LayerInfosForCopy` to validate against, instead.